File: Examples/test-suite/lua/director_nspace_runme.lua
require("director_nspace")

-- Create a derived class
local MyBarFoo = {}
function MyBarFoo:new()
    local parent = director_nspace.TopLevel.Bar.Foo
    self.__index = self
    -- Inherit TopLevel::Bar::Foo
    setmetatable(self, { __index = parent })
    -- Inside swig_derive() we can only call the partent!
    local pingFunc = function(self)
      return "director_nspace_MyBarFoo.ping()"
    end
	local obj = parent()
	-- Use SWIG-provided helper to set multiple overrides at once
	swig_derive(obj, {
      ping = pingFunc,
      pong = function(self)
        return "director_nspace_MyBarFoo.pong()" .. pingFunc(self)
        -- self:ping() Lua call original C++ class method
      end,
      fooBar = function(obj)
        return obj:FooBarDo()
      end,
      makeFoo = function()
        return director_nspace.TopLevel.Bar.Foo()
      end,
      makeFooBar = function()
        return director_nspace.TopLevel.Bar.FooBar()
      end
    })
	return obj
end

local c = MyBarFoo:new()
assert(c:ping() == "director_nspace_MyBarFoo.ping()")
assert(c:pong() == "director_nspace_MyBarFoo.pong()director_nspace_MyBarFoo.ping()")

File: Examples/test-suite/python/director_nspace_runme.py
from swig_test_utils import swig_check

import director_nspace


class PythonFoo(director_nspace.TopLevel.Bar.Foo):
    def ping(self):
        return "PythonFoo.ping();"


foo = PythonFoo()
swig_check(foo.ping(), "PythonFoo.ping();")
swig_check(foo.pong(), "Bar::Foo::pong();PythonFoo.ping();")
swig_check(foo.fooBar(director_nspace.TopLevel.Bar.FooBar()), "Bar::Foo2::Foo2Bar()")
swig_check(type(foo.makeFoo()), director_nspace.TopLevel.Bar.Foo)
